File: java/src/main/java/mx/iteso/privileges/Edit.java
package mx.iteso.privileges;

import mx.iteso.PrivilegeDecorator;
import mx.iteso.User;

import java.util.Set;

/**
 * Edit decorator class.
 */
public class Edit extends PrivilegeDecorator {
    /**
     * User attribute.
     */
    private User user;

    /**
     *
     * @param nUser user to be decorated
     */
    public Edit(final User nUser) {
        this.user = nUser;
        this.user.getPrivileges().add("edit");
    }

    /**
     *
     * @return privileges of the user
     */
    public Set<String> getPrivileges() {
        return user.getPrivileges();
    }

    /**
     *
     * @return has publish privilege
     */
    public boolean publish() {
        return user.publish();
    }

    /**
     *
     * @return has edit privilege
     */
    public boolean edit() {
        return user.getPrivileges().contains("edit");
    }

    /**
     *
     * @return has delete privilege
     */
    public boolean delete() {
        return user.delete();
    }

    /**
     *
     * @return has approve privilege
     */
    public boolean approve() {
        return user.approve();
    }
}

File: python/src/main/python/Edit.py
from PrivilegeDecorator import *

class Edit(PrivilegeDecorator):
    def __init__(self, oldUser):
        self.user = oldUser
        self.user.getPrivileges()["edit"] = "edit"

    def getPrivileges(self):
        return self.user.getPrivileges()

    def publish(self):
        return self.user.publish()

    def delete(self):
        return self.user.delete()

    def approve(self):
        return self.user.approve()

    def edit(self):
        if "edit" in self.user.getPrivileges():
            return True
        else:
            return False